### PR TITLE
Fix slot skipping (install-same) handling twice

### PR DESCRIPTION
**Pull request by ejoerns**
This fixes slot skipping handling on two levels.

With v1.10, slot skipping got broken while with v1.11 the slot skipping *detection* got broken (and thus hides the previously introduced issue).

The error that will be raised by v1.10 is:

```
Not enough substeps: update_slots (12/20)
```

Fix both and add a minimal test case that triggers an installation twice.

### CODE REVIEW

File: test/install.c [hunk 1]
```diff
@@ -1124,6 +1141,20 @@ static void install_test_bundle(InstallFixture *fixture,
...
 
+static void install_test_bundle_twice(InstallFixture *fixture,
+		gconstpointer user_data)
+{
+	/* First run */
+	install_test_bundle(fixture, user_data);
+
+	/* Manual partial cleanup */
+	test_umount(fixture->tmpdir, "mnt");
+	test_rm_tree(fixture->tmpdir, "mnt");
```
Comment (jluebbe) [inline test/install.c:1152]: `install_test_bundle()` should already handle the umount. Why is this needed?

File: test/install.c [hunk 2]
```diff
@@ -1124,6 +1141,20 @@ static void install_test_bundle(InstallFixture *fixture,
...
+
+	/* Second run */
+	install_test_bundle(fixture, user_data);
```
Comment (jluebbe) [inline test/install.c:1155]: Can we check that the install was skipped? Perhaps by looking at the slot status data?

Also we should test the inverse: that the bundle is installed again if install-same is true.

File: test/test-install-same.conf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# testsuite system configuration
```
Comment (jluebbe) [inline test/test-install-same.conf:1]: This should perhaps be named `test/test-install-same-false.conf`

File: test/test-install-same.conf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+[slot.appfs.0]
+device=images/appfs-0
+type=ext4
+parent=rootfs.0
+
+[slot.appfs.1]
+device=images/appfs-1
+type=ext4
+parent=rootfs.1
+
```
Comment (jluebbe) [inline test/test-install-same.conf:45]: If these are not needed, please remove them.

**Review by jluebbe — CHANGES_REQUESTED** [review]

File: test/install.c (hunk 1 above)
Comment (ejoerns) [reply test/install.c:1152]: In fact, only the removal of the folder is needed. But this could be done by `install_test_bundle()` itself.

File: test/install.c (hunk 2 above)
Comment (ejoerns) [reply test/install.c:1155]: I don't see any way to check this with manageable effort at the moment or in the context of this fix.

Since we use the same bundle for installing, the only way to detect if an installation happened is by having a look at the time stamps in the status file. And I am not sure if enough time is spent between calls.

Anyway, adding support for parsing the status file here is quite complex from the layer we operate on.

With the current check, at least the older issue would have been detected. The other would have not.

File: test/test-install-same.conf (hunk 3 above)
Comment (ejoerns) [reply test/test-install-same.conf:1]: Will rename
